app/speak_slack.rb

```diff
@@ -0,0 +1,15 @@
+# Takes an array of food truck names and formats it to slack
+require_relative '../constants'
+
+module FoodTruck
+  class SpeakSlack
+    def self.trucks(trucks: [])
+      msg = ""
+      trucks.uniq!
+      trucks.each do |truck|
+        msg << "\n#{truck}"
+      end
```

Comment:
This logic relies extensively on mutating things:

- `trucks` is mutated via the `trucks.uniq!` call which means this method both returns a value and changes the array passed by the caller.  That's usually a bad thing; it leads to surprises if the caller does something like this:

``` ruby
trucks = get_trucks_from_somwhere(...)
slack_format = FoodTruck:: SpeakSlack.speak_slack(trucks)
puts "Formatted original #{trucks.inspect} into slack format: #{slack_format}"
```

Here the caller is assuming `speak_slack` is only returning a value and not also mutating the arg (why would it expect the arg to be mutated, after all?).

`msg` is also being mutated throughout.  I'd recommend this instead:

``` ruby
if trucks.empty?
  MSG_NO_FOOD_TRUCKS
else
  trucks.uniq.join("\n")
end
```